rust/src/wrapper/device/mod.rs

```diff
@@ -94,7 +120,10 @@ impl Device {
             PyModule::import(py, intern!(py, "bumble.device"))?
                 .getattr(intern!(py, "Device"))?
                 .call((), Some(kwargs))
-                .map(|any| Self(any.into()))
+                .map(|any| Self {
+                    obj: any.into(),
+                    advertising_status: AdvertisingStatus::NotAdvertising,
```

Comment:
what if the device is a real dongle and it's actually already advertising?

Reply:
The device could pretty much be in any state (beyond just advertising/not advertising) until we send HCI_Reset during `power_on`.

We could (a) prevent some operations if `power_on` has not been called (b) reset the controller during `__init__` to ensure a clean slate (c) leave as-is

I wrote this to match how its written in `device.py`, though I do think (b) would be better (with some parameter to disable reset if desired).

(a) could be fine, but could result in more invalid states by accident if not careful

(b) would ideally be implemented in `device.py` since I don't think it makes sense to fix the bug in Rust only, but am curious to hear thoughts from @barbibulle